services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/registry.py

```diff
@@ -46,47 +50,94 @@ async def _is_registry_reachable(registry_settings: RegistrySettings) -> None:
             raise _RegistryNotReachableError(error_message)
 
 
-async def wait_for_registry_liveness(app: FastAPI) -> None:
-    registry_settings: RegistrySettings = app.state.settings.REGISTRY_SETTINGS
+async def _is_dockerhub_reachable(registry_settings: RegistrySettings) -> None:
+    async with httpx.AsyncClient(timeout=5) as client:
+        params: dict[str, Any] = {"headers": {"Content-Type": "application/json"}}
+        if registry_settings.REGISTRY_AUTH:
+            params["auth"] = (
+                registry_settings.REGISTRY_USER,
+                registry_settings.REGISTRY_PW.get_secret_value(),
+            )
+            params["data"] = json.dumps(
+                {
```

Comment:
Q: about actually both of the registry liveness checks. I think you mostly use docker to connect in the end right? why not just use docker to do the checking instead of curl'ing? Also this would spare the need to add the docker registry URL.